CONVENTIONS.md

```diff
@@ -29,6 +29,85 @@ Black is left at defaults, flake8 and isort are configured to adhere to black.
 
 All functions need to be covered by unit tests.
 
+# Coordinate frame conventions
+
+As the purpose of pylinalg as a library primarily is to support linear algebra
+operations in pygfx and applications based on pygfx, we adhere to a number of
+standard coordinate frame conventions to align with expectations from pygfx.
+
+To start with, we identify three important coordinate frames that we use as
+references:
+
+1. **world**: The world frame is the (global) inertial reference frame. All
+    other coordinate frames are positioned relative to *world*; either
+    explicitly or implicitly by being placed relative to a sequence of frames
+    that were previously placed relative to *world*. This creates a graph of
+    coordinate frames (called the scene graph), whith *world* at its root and it
+    allows finding a transformation matrix between pairs of coordinate frames.
+2. **local**: The local frame is the coordinate frame in which an object's
+    vertices are expressed. For example, when inspecting the position of a
+    cube's corners, then the numerical values of the corners are given in the
+    cube's *local frame*.
+3. **parent**: The parent frame is the coordinate frame in which an object's
+   pose (position + orientation) are expressed. *Parent* can either be the
+   inertial reference frame (world) or it can be another object's local frame.
+   An object's position is always relative to its parent; for example, if a
+   lightbulb has a lamp's local frame as it's parent, then the lightbulb's
+   position expressed in world coordinates will change whenever the position of
+   lamp changes, i.e., if the lamp moves, so does the lightbulb.
+
+Further, when talking about transformation matrices and coordinate transforms,
+we use two additional frames to avoid confusion:
+
+* **source**: The source frame is the coordinate frame in which
+to-be-transformed vectors are expressed, i.e., it is the reference frame of the
+input vectors.
+* **target**: The target frame is the coordinate frame in which transformed
+  vectors are expressed, i.e., it is the reference frame of the output vectors.
+
+Generally there are four transform steps when going from an object's local space,
+all the way to screen space:
+
+* Local space to world space: world matrix
+* World space to view space: view matrix
+* View space to NDC/clip space: projection matrix
+
+The view and projection matrices are owned by camera objects. The view
+matrix can be acquired by inverting the camera's world matrix. The projection matrix
+is configured independently on the camera and depends on its
+type (perspective, orthographic).
```

Comment:
```suggestion
To render an object we have to express it's vertices in a camera's so-called NDC
coordinates, which stands for normalized device coordinates. Recall that
vertices are expressed in the object's *local* frame, which means we need to
work out the transformation from object *local* to NDC. We can do this by
following the chain of transformations in the scene graph from object *local*
via *world* to camera *local* and from camera *local* into NDC. Since this chain
is very important for rendering, it's parts have special names:

* **World Matrix**: The transformation from object's *local* into *world*.
* **View Matrix**: The transformation from *world* to camera's *local*. This is
  the inverse of the camera's world matrix.
* **Projection Matrix**: The transformation from camera's *local* into NDC.
```

We can introduce this by connecting to the frame conventions we used above. Also, having written this, would it flow better if this gets mentioned after the axis naming convention?